code/game/objects/items/mail_ch.dm

```diff
@@ -145,8 +134,7 @@
 
 	var/datum/job/this_job = SSjob.name_occupations[new_recipient.job]
 	if(this_job)
-		if(SSjob.get_primary_department_of_job(this_job) && department_colors[SSjob.get_primary_department_of_job(this_job)])
-			color = department_colors[SSjob.get_primary_department_of_job(this_job)]
+		color = SSjob.get_primary_department_of_job(this_job).color
```

Comment:
SSjob.get_primary_department_of_job(this_job) runs a proc, so adding .color to the end causes dreamchecker to fail. You'll probably want something like var/relevant_typepath = SSjob.get_primary_department_of_job(this_job) then take color from that var (assuming the proc works that way).

Reply:
Should be solved now